src/device/gcn/output.jl

```diff
@@ -174,7 +194,10 @@ function Base.unsafe_load(ptr::LLVMPtr{ROCPrintfBuffer,as} where as)
         ptr += sizeof(UInt64)
         UInt64(T_ptr) == 0 && break
         T = unsafe_pointer_to_objref(T_ptr)
-        arg = unsafe_load(convert(Ptr{T}, ptr))
+        valid, arg = semi_safe_load(convert(Ptr{T}, ptr))
+        if !valid
+            @warn "@rocprintf: Memory read failed!\nFuture read failures will be ignored" maxlog=1
+        end
         push!(args, arg)
```

Comment:
```suggestion
        arg = arg::T
        push!(args, arg)
```

Reply:
This will of course fail if we get `nothing`. I'm thinking we could return a zero-initialized object of the appropriate type (which must necessarily be bitstype), and indicate in the warning that printed data may be incorrect.